test/core.jl

```diff
@@ -7508,3 +7508,15 @@ let array = Int[]
 end
 @test compare_union37557(Ref{Union{Int,Vector{Int}}}(1),
                          Ref{Union{Int,Vector{Int}}}(1))
+
+# @aggressive_constprop ======
+g_nonaggressive(y, x) = (for i = 1:100; end; Val{x}())
```

Comment:
Rather than the suprious loop, would it be clearer to just mark these as `@noinline`?

Reply:
I thought I had tried that and it didn't work, but I guess I messed up somehow. Will do that instead.